spec/abstractops.html

```diff
@@ -1507,7 +1507,6 @@ <h1>ParseTemporalYearMonthString ( _isoString_ )</h1>
       1. Assert: Type(_isoString_) is String.
       1. If _isoString_ does not satisfy the syntax of a |TemporalYearMonthString| (see <emu-xref href="#sec-temporal-iso8601grammar"></emu-xref>), then
         1. Throw a *RangeError* exception.
-      1. If ! SameValue(_isoString_, *"-000000"*) is *true*, throw a *RangeError* exception.
```

Comment:
why is this nonsensical?

Reply:
please read the commit message

Reply:
in that case, it seems clearer to me to move this check up, rather than removing it, since "satisfy the syntax" is a more difficult condition to understand.

Reply:
iow, it's not nonsensical, it's just redundant, but the redundancy is actually helpful.

Reply:
Actually I think this is a mistake, the check should examine `_result_.[[Year]]` and not `_isoString_`

Reply:
I don't see how that's helpful, the same check will be done either way. `-000000` on its own is never a valid `isoString`, the intention here is to filter out a single year the production considers valid, so having a random check of the whole input string against a single invalid year makes no sense.

Reply:
Moving it up is not even a possibility, and by the time we have the `result` that very same check has already been done via `ParseISODateTime`. I genuinely don't get the fuss here.

Reply:
I think you're right, this was introduced in #1992 but I didn't clock that ParseISODateTime was called below when I reviewed it. I think it's OK to remove this line. cc @jessealama

Reply:
> "satisfy the syntax" is a more difficult condition to understand

@ljharb re. this, it's on my and @gibson042's radar to replace these vague "satisfy the syntax" conditions with calls to the ParseText operation.

Reply:
@ptomato that's fine, but "ParseText" is still less clear than explicitly handling edge cases like this as their own steps.

Reply:
I agree (and that sentiment shared between me and @jessealama is why he wrote it that way in the first place), but on the other hand it seems annoying for implementors if we have to make a bunch of parsing operations fallible. It might be as easy as changing the grammar to say something like

_DateExtendedYear_ : _Sign_ _DecimalDigit_ _DecimalDigit_ _DecimalDigit_ _DecimalDigit_ _DecimalDigit_ _DecimalDigit_ but not **-000000** or **\u2212 000000**

which I think is still pretty clear. And if I'm wrong about that and it turns out to be something totally impenetrable, we can always look for a different solution that remains clear and readable.